Additional details on contracts page

mhausberg opened the pull request:
As discussed with @TyKonKet I integrated my w.i.p. on the ingame contracts menu page with this BetterContracts mod.

Commit:
1st integration

integrated my gui in BetterContracts 1.1.00

Commit:
ready for testing

minor updates to mod description.
ready for testing

mhausberg commented on src/modDesc.xml:
```diff
@@ -30,19 +34,23 @@ Changelog ${version}:
-Dieser Mod hilft das Vertragssystem zu verbessern, sowohl im Einzelspieler und Multiplayer.
+"Mach ich jetzt diese drei Düngemissionen, oder doch lieber den 2.3 ha Kartoffelernte-Vertrag?" "Wieviel Flüssigdünger brauche ich für diesen Auftrag?" Falls Sie sich je derartige Fragen gestellt haben, hilft Ihnen dieser Mod bei den Antworten. Er verbessert das Vertragsmanagement, sowohl im Einzelspieler- als im Multiplayer-Modus.
 
-- Die maximale Anzahl und die Anzahl der generierten Missionen werden jetzt automatisch an die Anzahl der Felder in der Karte angepasst.
+- Die Maximalzahl sowie die Anzahl der jeweils neu generierten Missionen werden automatisch an die Zahl der Felder auf der Karte angepasst.
  
-- Sie können jetzt über die Schaltfläche "Neue Verträge" neue Missionen generieren und diese sogar alle auf dem Vertragsverwaltungsbildschirm löschen.
+- Sie können über die Schaltfläche "Neue Verträge" sofort neue Missionen generieren oder alle mit dem Button "Liste löschen" entfernen.
 
-- Verträge werden jetzt nach Typ und Feldnummer sortiert, um das Auffinden des gewünschten Vertrags zu erleichtern.
+- Mit dem "Details" Button kann die Anzeige der zusätzlichen Vertragsdetails ein- und ausgeschaltet werden.
+
+- Sie können die Vertragsliste sortieren nach Vertragstyp/ Feldnummer, nach Gesamtprofit, und nach Profit pro Minute, um einen gewünschten Vertrag schnell zu finden.
  
 - Sie können mehr als einen Vertrag gleichzeitig aktivieren.
  
-- Die Feldnummernanzeige (auf der Minikarte) der Felder, auf denen Sie Verträge aktivieren, beginnt zu blinken, um Ihnen den richtigen Weg zu zeigen.
+- Die Feldnummernanzeigen (auf der Minikarte) der Felder mit Ihren aktiven Verträgen blinken, um Ihnen den Weg zu zeigen.
 
 Changelog ${version}:
+- Zusätzliche Details: Neben Feldnummer, Feldgröße, Auftragsart werden weitere Details des Auftrags angezeigt. Z.B. die geschätzten Kosten des Verbrauchsmaterials (Dünger, Pflanzenschutzmittel, Saatgut). Bei Ernte- und Gras-Missionen wird die abzuliefernde Mindestmenge angezeigt, sowie die Erntemenge, die Sie behalten (und verkaufen) können. Daraus ermittelt der Mod den Gesamtertrag der Mission,  ...
+
 - Konfliktverhütung mit anderen Mods hinzugefügt
 - Sortierung der Vertragslisten hinzugefügt
 - Es wurden viele neue Fahrzeugkombinationen für Verträge hinzugefügt
```
updated only English/ German. Other languages need to be done.

Lactic68 replied:
I will do the french translation when this will be merged into the main project.

mhausberg commented on src/modDesc.xml:
```diff
@@ -120,4 +131,16 @@ Changelog ${version}:
+        <actionBinding action="MENU_EXTRA_3">
+            <binding device="KB_MOUSE_DEFAULT" input="KEY_s" index="1"/>
+        </actionBinding>
+        <actionBinding action="SC_Con" >
+            <binding device="KB_MOUSE_DEFAULT" input="KEY_lshift KEY_c" />
+        </actionBinding>
```
only used when debug == true. Will print info on all avaiablel missions to the log

TyKonKet replied:
Can you please remove this?
I know it's only used if `debug == true` but it will always register the actions and bindings on every user `inputBinding.xml` so that they cannot use the same keyboard keys for anything else.
Try using a console commands instead.
`addConsoleCommand`

mhausberg replied:
good hint - addConsole Command was much easier than going via an action key!

TyKonKet commented on src/betterContracts.lua:
```diff
@@ -4,35 +4,116 @@
+        {"sortpmin", g_i18n:getText("SC_sortpMin")}, 
+    }
+
+    -- make my Gui texts global, needed for my Gui profilesd to work
```
This isn't really needed since che `RoyalMod.new()` already do it

TyKonKet commented on src/modDesc.xml:
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
-<modDesc descVersion="${descVersion}">
+<modDesc descVersion="52">
```
Yes, don't worry

TyKonKet commented on src/scripts/gui.lua:
```diff
@@ -0,0 +1,350 @@
+    inGameMenu.newContractsButton = nil
+    inGameMenu.clearContractsButton = nil
+    inGameMenu.detailsButton = nil
+end;
```
```suggestion
end
```

TyKonKet commented on src/scripts/gui.lua:
```diff
@@ -0,0 +1,350 @@
+    	inGameMenu.detailsButton.onClickCallback(inGameMenu)
+		inGameMenu:playSample(GuiSoundPlayer.SOUND_SAMPLES.CLICK)
+    end
+end;
```
```suggestion
end
```

TyKonKet reviewed (changes requested):
Hi @mhausberg 
The most important things for us are mainly two:
 - Code formatting
 - Consistency in variables and functions naming
I've added some comments to show you some of that small issues, otherwise everything seems fine.
Go ahead with those fixes, I'll do a second and hopefully final review after.

fcelsa commented on src/modDesc.xml:
```diff
@@ -120,4 +131,16 @@ Changelog ${version}:
 			<ru>Очистить контракты</ru>
 		</text>
```
l10n texts needs to be moved to translations files.

Commit:
Update src/betterContracts.lua

Co-authored-by: Matteo Autieri <matteoautieri@live.it>

Commit:
Update src/betterContracts.lua

Co-authored-by: Matteo Autieri <matteoautieri@live.it>

Commit:
review changes

I understand and appreciate all suggested changes - thank you!

Commit:
resolved upstream change conflict

I tried to commit from 2 of the suggested changes (on the web), and also some of my local edits..
